### PR TITLE
[WIP] Add retries to subscription storage operation

### LINKED ISSUE #232: Gracefully handle concurrency errors with scaled-out subscribers
Scaled-out subscribers may cause concurrency errors (PK violation) during processing of subscription messages by the publisher.

These errors should be gracefully handled within the persister, as it is currently done RavenDB and ASP persisters:

https://github.com/Particular/NServiceBus.Persistence.AzureStorage/blob/develop/src/NServiceBus.Persistence.AzureStorage/Subscriptions/AzureSubscriptionStorage.cs#L62

https://github.com/Particular/NServiceBus.RavenDB/blob/develop/src/NServiceBus.RavenDB/Subscriptions/SubscriptionPersister.cs#L31

The faulty tests have all been moved to the `Acceptance tests - quarantine build configuration`.

### PR DESCRIPTION
**Pull request by MarcinHoppe**
Fixes #232

### CODE REVIEW

**Comment by MarcinHoppe** [comment]
@Particular/nhibernate-persistence-maintainers Please review and merge.

File: src/NServiceBus.NHibernate/Subscriptions/SubscriptionPersister.cs [hunk 1]
```diff
@@ -25,6 +27,37 @@ public SubscriptionPersister(ISessionFactory sessionFactory)
...
+                    }
+                    else
+                    {
+                        // This was the last attempt, give up.
+                        throw;
+                    }
+                }
+            }
+
+            throw new Exception($"Internal error occured when storing subscription of endpoint '{subscriber.Endpoint}' to message '{messageType}'.");
```
Comment (SzymonPobiega) [inline src/NServiceBus.NHibernate/Subscriptions/SubscriptionPersister.cs]: When do we get here?

Comment (MarcinHoppe) [inline src/NServiceBus.NHibernate/Subscriptions/SubscriptionPersister.cs]: We should never get here. If we do get here, it's a bug.

**Comment by SzymonPobiega** [comment]
I think (but I am not 100% sure) we need to add similar thing for the unsubscribe because `Delete` in NH expects 1 row affected. In the scale-out scenario we might have multiple attempts to delete the same row, only one of which will succeed.

**Comment by MarcinHoppe** [comment]
@SzymonPobiega Doesn't [this loop](http://infoshareacademy.com/zostan_trenerem.html) delete `Subscription` records one-by-one?

**Comment by SzymonPobiega** [comment]
@MarcinHoppe still working on this?

**Comment by MarcinHoppe** [comment]
@SzymonPobiega Yes, I'm still on it. ATTs fail regularly on TeamCity (but Work on My Machine (TM)) and I am trying to investigate why that is.

**Pull request closed without merging**